src/genie/libs/parser/hp_comware/show_interface.py

```diff
@@ -0,0 +1,703 @@
+                out_late_collision = m.groupdict()['out_late_collision']
+
+                if(out_abort == '-'):
+                    out_abort = 0
+                if(out_deferred == '-'):
+                    out_deferred = 0
+                if(out_collision == '-'):
+                    out_collision = 0
+                if(out_late_collision == '-'):
+                    out_late_collision = 0
+
+                interface_dict[interface]['counters']["out_abort"] = int(out_abort)
+                interface_dict[interface]['counters']["out_deferred"] = int(out_deferred)
+                interface_dict[interface]['counters']["out_collision"] = int(out_collision)
+                interface_dict[interface]['counters']["out_late_collision"] = int(out_late_collision)
+                continue
+
+            # '^ *(?P<out_lost_carrier>.*) lost carrier, (?P<out_no_carrier>.*) no carrier$'
+            m = p28.match(line)
+            if m:
+                out_lost_carrier = m.groupdict()['out_lost_carrier']
+                out_no_carrier = m.groupdict()['out_no_carrier']
+
+                if(out_lost_carrier == '-'):
+                    out_lost_carrier = 0
+                if(out_no_carrier == '-'):
+                    out_no_carrier = 0
+
+                interface_dict[interface]['counters']["out_lost_carrier"] = int(out_lost_carrier)
+                interface_dict[interface]['counters']["out_no_carrier"] = int(out_no_carrier)
+                continue
+
+        return(interface_dict)
+
+
+class DisplayIpInterfaceBriefSchema(MetaParser):
+    """Parser for display ip interface brief"""
+    schema = {
+            Optional('route'):
+            {
+                Any():
+                {
+                    Optional('ip_address'): str,
+                    Optional('link'): str,
+                    Optional('protocol'): str,
+                    Optional('description'): str
+                }
+            },
+            Optional('bridge'):
+            {
+                Any():
+                {
+                    Optional('link'): str,
+                    Optional('speed'): str,
+                    Optional('duplex'): str,
+                    Optional('type'): str,
+                    Optional('pvid'): str,
+                    Optional('description'): str
+                }
+            }
+    }
+
+
+class DisplayInterfacesBrief(DisplayIpInterfaceBriefSchema):
+    """parser for display interface brief
+    """
+    ##########################################################
+    #      Other Plataforms equivalent command
+    ##########################################################
+    platform_equivalent_cli = ['show interface brief']
+    ##########################################################
+    #      HP Comware command
+    ##########################################################
+    platform_cli = ['display interface brief']
+
+    cli_command = platform_equivalent_cli + platform_cli
```

Comment:
Do you need this?

Reply:
I created that structure for two reasons:
1. Create flexibility for the end user in multi vendor environments 
2. Improve readability

For the first reason in multi vendor scenario the user might need to add some logic to check the device platform and choose the appropriate command, with that the parser accepts equivalents commands

For the seccond reason, I could achieve the same result by making a big list with the platform command and the cisco equivalent, but i think that way the code is cleaner.

@ahersi2-cisco what do you think keep the structure or use the simple cli_command = ['display interface brief'] ?

Reply:
I see, if you foresee that scenario happening, having it setup in this manner is no harm